counsel.el

```diff
@@ -221,7 +221,8 @@ Update the minibuffer with the amount of lines collected every
                              lines)
              lines))))
       (let ((ivy--prompt (format "%d++ %s" numlines (ivy-state-prompt ivy-last))))
-        (ivy--insert-minibuffer (ivy--format ivy--all-candidates)))
+        (with-selected-window (active-minibuffer-window)
```

Comment:
Is the minibuffer guaranteed to be active? Wouldn't it be safer to check for `nil`, just in case?

```diff
diff --git a/counsel.el b/counsel.el
index 102deaf..3264fae 100644
--- a/counsel.el
+++ b/counsel.el
@@ -220,8 +220,11 @@ counsel--async-filter
                                (string-match-p counsel-async-ignore-re line))
                              lines)
              lines))))
-      (let ((ivy--prompt (format "%d++ %s" numlines (ivy-state-prompt ivy-last))))
-        (ivy--insert-minibuffer (ivy--format ivy--all-candidates)))
+      (let ((ivy--prompt (format "%d++ %s" numlines (ivy-state-prompt ivy-last)))
+            (win (active-minibuffer-window)))
+        (when win
+          (with-selected-window win
+            (ivy--insert-minibuffer (ivy--format ivy--all-candidates)))))
       (setq counsel--async-time (current-time)))))
 
 (defun counsel-delete-process (&optional name)
```

Reply:
If we want to check for that I think we should not ignore the nil case but fail instead.

Reply:
What purpose would throwing an error in a filter function serve, other than potentially annoying the user? Can you guarantee that there will never be a race condition between Emacs accepting asynchronous process output and the user exiting the minibuffer?

Reply:
I think that if we can't write to the minibuffer, the results will be incomplete and we should notify the user.

You are probably more familiar with async process programming in elisp than I am. If I want to log this kind of error without failing, what are my options? I failed to find information about it.